testing/reconciler.go

```diff
@@ -214,14 +214,18 @@ func (tc *ReconcilerTestCase) Run(t *testing.T, scheme *runtime.Scheme, factory
 		ExpectStatusPatches:     tc.ExpectStatusPatches,
 	}
 
+	c := expectConfig.Config()
+	ctx = reconcilers.StashConfig(ctx, c)
+	ctx = reconcilers.StashOriginalConfig(ctx, c)
```

Comment:
The config should not be stashed by the test harness because that's the ResourceReconciler's responsibility.

The root cause of the issue here is that the Validate method is called directly as part of the test. At runtime, Validate is called within the SetupWithManager method, and the context is configured there.

https://github.com/reconcilerio/runtime/blob/ec87995ce1046940d5c466110ed84975387d0fcb/reconcilers/resource.go#L179-L184

I think the right approach here is to also setup the context when the entry point is the Validate method.

Created #654

Reply:
That makes sense. I'll close this PR.